src/hotspot/share/opto/vectorIntrinsics.cpp

```diff
@@ -2493,13 +2494,18 @@ bool LibraryCallKit::inline_vector_convert() {
     if (is_mask && is_floating_point_type(elem_bt_from)) {
       new_elem_bt_from = elem_bt_from == T_FLOAT ? T_INT : T_LONG;
     }
+    if (is_mask && is_floating_point_type(elem_bt_to)) {
+      new_elem_bt_to = elem_bt_to == T_FLOAT ? T_INT : T_LONG;
+    }
     int cast_vopc = VectorCastNode::opcode(new_elem_bt_from, !is_ucast);
-    // Make sure that cast is implemented to particular type/size combination.
-    if (!arch_supports_vector(cast_vopc, num_elem_to, elem_bt_to, VecMaskNotUsed)) {
+
+    // Make sure that vector cast is implemented to particular type/size combination.
+    bool no_vec_cast_check = is_mask && (src_type->isa_vectmask() ||
```

Comment:
To make it safer, let's also check `dst_type->isa_vectmask()` like the following.
```
      if (is_mask) {
        if ((dst_type->isa_vectmask() && src_type->isa_vectmask()) ||
            (type2aelembytes(elem_bt_from) == type2aelembytes(elem_bt_to))) {
          op = gvn().transform(new VectorMaskCastNode(op, dst_type));
        } else {
          op = VectorMaskCastNode::makeCastNode(&gvn(), op, dst_type);
        }
      }
```

Reply:
From the check https://github.com/openjdk/jdk/blob/master/src/hotspot/share/opto/vectorIntrinsics.cpp#L2485, we know the basic mask type of dst and src must be the same. So just checking `src_type` is ok here. Besides, this part of changes will be removed finally after I unify the vector mask cast operation.

Reply:
But the mismatch seems strange to me.

Maybe, you can 
1) Add `dst_type->isa_vectmask()` check here, or
2) Also delete `dst_type->isa_vectmask()`  (https://github.com/openjdk/jdk/blob/master/src/hotspot/share/opto/vectorIntrinsics.cpp#L2551)

Reply:
I'v added the dst_type check in the latest commit. Thanks!